ios/FluentUI/SegmentedControl/SegmentedControl.swift

```diff
@@ -373,32 +370,33 @@ open class SegmentedControl: UIView, TokenizedControlInternal {
             }
         }
 
-        let windowSafeAreaInsets = window.safeAreaInsets
-        let windowWidth = window.bounds.width - windowSafeAreaInsets.left - windowSafeAreaInsets.right
         if isFixedWidth {
-            if traitCollection.userInterfaceIdiom == .pad {
-                width = max(windowWidth / 2, Constants.iPadMinimumWidth)
-            } else {
-                width = windowWidth
+            if let window = window {
+                let windowSafeAreaInsets = window.safeAreaInsets
+                let windowWidth = window.bounds.width - windowSafeAreaInsets.left - windowSafeAreaInsets.right
+                if traitCollection.userInterfaceIdiom == .pad {
+                    width = max(windowWidth / 2, Constants.iPadMinimumWidth)
+                } else {
+                    width = windowWidth
+                }
             }
         } else {
             width += (contentInset.leading + contentInset.trailing)
         }
         height += (contentInset.top + contentInset.bottom)
 
-        let finalWidth = min(min(width, windowWidth), size.width)
-        let finalHeight = min(height, size.height)
-        return CGSize(width: finalWidth,
-                      height: finalHeight)
+        return CGSize(width: min(width, size.width),
+                      height: min(height, size.height))
     }
 
     open override func didMoveToWindow() {
         super.didMoveToWindow()
 
         tokenSet.update(fluentTheme)
-        updateColors()
-        updateButtons()
```

Comment:
Why are the update calls removed?

Reply:
Because they're in `tokenSetSink`, so if the theme changes they'll get updated. We don't need to update the colors twice.